diff --git a/dist/node/chunks/dep-ByPKlqZ5.js b/dist/node/chunks/dep-ByPKlqZ5.js
index 23b674848a4639aed3c6866373eaf3541d71d55f..329b51f95e40737d48e16b6ea93a8e83046d6e02 100644
--- a/dist/node/chunks/dep-ByPKlqZ5.js
+++ b/dist/node/chunks/dep-ByPKlqZ5.js
@@ -48433,7 +48433,7 @@ function cssPostPlugin(config) {
         return `export default ''`;
       }
       const inlined = inlineRE.test(id);
-      const modules = cssModulesCache.get(config).get(id);
+      const modules = cssModulesCache.get(config)?.get(id);
       const modulesCode = modules && !inlined && dataToEsm(modules, { namedExports: true, preferConst: true });
       if (config.command === "serve") {
         const getContentWithSourcemap = async (content) => {
